From 91118d84995c9b13f1a30b63ed3ece017dd8b04a Mon Sep 17 00:00:00 2001
From: Laurent Bigonville <bigon@bigon.be>
Date: Mon, 15 Oct 2018 10:37:42 +0200
Subject: [PATCH] build: Set absolute path of libexecdir

get_option('libexecdir') contains the path to libexecdir WITHOUT the
prefix, we need to prepend the correct prefix to the expanded path is
correct
---
 data/meson.build | 2 +-
 demo/meson.build | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index 3fdba43..b5b48cc 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -14,7 +14,7 @@ if get_option('enable-backend')
                    install_dir: conf_dir)
 
     conf = configuration_data()
-    conf.set('libexecdir', get_option('libexecdir'))
+    conf.set('libexecdir', join_paths(get_option('prefix'), get_option('libexecdir')))
     conf.set('dbus_srv_user', get_option('dbus-srv-user'))
 
     service_dir = get_option('datadir') + '/dbus-1/system-services'
diff --git a/demo/meson.build b/demo/meson.build
index 07d4797..6e28709 100644
--- a/demo/meson.build
+++ b/demo/meson.build
@@ -3,7 +3,7 @@ demo_dir = get_option('libexecdir') + '/geoclue-2.0/demos'
 
 i18n = import('i18n')
 desktop_conf = configuration_data()
-desktop_conf.set('libexecdir', get_option('libexecdir'))
+desktop_conf.set('libexecdir', join_paths(get_option('prefix'), get_option('libexecdir')))
 
 if get_option('libgeoclue')
     include_dirs = [ configinc,
-- 
2.18.1

